setup.py

```diff
@@ -111,8 +111,8 @@
     # We use numpy>=1.17 to have np.random.Generator (Dataset shuffling)
     "numpy>=1.17",
     # Backend and serialization.
-    # Minimum 8.0.0 to be able to use .to_reader()
-    "pyarrow>=8.0.0",
+    # Minimum 14.0.1 to fix vulnerability CVE-2023-47248
+    "pyarrow>=9.0.0",  # TODO: maximum version allowed by Apache Beam
```

Comment:
We can require `pyarrow-hotfix<1` and `import pyarrox_hotfix` to fix `CVE-2023-47248` without requiring `14.0.1`